samples/extensions/shader_object/shader_object.cpp

```diff
@@ -423,6 +421,7 @@ void ShaderObject::load_assets()
 	VK_CHECK(vkCreateSampler(get_device().get_handle(), &sampler_create_info, nullptr, &heightmap_texture.sampler));
 
 	// Setup a repeating sampler for the terrain texture layers
+	vkDestroySampler(get_device().get_handle(), terrain_array_textures.sampler, nullptr);
```

Comment:
Is this now leaked?

Reply:
Can you elaborate? The sample did leak some stuff, and e.g. this change makes sure all resources are properly freed. I made sure that I didn't get any validation errors at instance destruction.